src/dataexplorer/dataexplorer-application/src/components/Patient/PatientQuestionnaireResponses/PatientQuestionnaire/PatientView/PatientView.tsx

```diff
@@ -39,11 +44,25 @@ const PatientView = (props: {
         },
     ];
 
-    const handleClick = (record: IRecord) => {
-        setSchemanumber(record.id);
+    const handleClick = (record: IDataSource) => {
+        record.id && setSchemanumber(record.id);
         history.push('/pasient/skjema');
     };
 
+    const handleCompareClick = () => {
+        setComparableSchemaNumbers(comparingSchemesIds);
+        history.push('pasient/skjema-sammenligning');
+    };
+
+    const addComparingSchemeOnRowClick = (id: ReactText) => {
+        setReachedMaxValue(false);
+        setComparingSchemesIds((comparingQuestionnaires) => [...comparingQuestionnaires, id]);
+    };
+
+    useEffect(() => {
+        if (reachedMaxValue && comparingSchemesIds.length === 2) message.warning('Maximum to skjemaer', 5);
+    }, [reachedMaxValue, comparingSchemesIds.length]);
```

Comment:
Is the double check of "reachedMaxValue" and ".. .length ===2" necessary? Should also maybe change the duration of the warning to something like 3.5, as it persists.

Reply:
Actually it did prevent a bug, but will fix the bug instead of checking it twice! 🍶